unit-tests/src/test/scala/scala/scalanative/unsafe/PtrBoxingSuite.scala

```diff
@@ -182,4 +182,31 @@ object PtrBoxingSuite extends tests.Suite {
       assert(out.toList == List(10, 20, 30))
     }
   }
+
+  test("load and store CFuncPtr") {
+    type GetInt       = CFuncPtr0[Int]
+    type StringLength = CFuncPtr1[CString, CSize]
+    type Functions    = CStruct2[GetInt, StringLength]
+    val getInt: GetInt = new GetInt {
+      override def apply(): CInt = 42
+    }
+    val stringLength = new StringLength {
+      override def apply(arg1: CString): CSize =
+        libc.string.strlen(arg1)
+    }
+    Zone { implicit z =>
+      val x = stackalloc[Functions]
+      x._1 = getInt
+      x._2 = stringLength
+
+      val loadedGetInt       = x._1
+      val loadedStringLength = x._2
+
+      val testStr = toCString("hello_native")
+      assertEquals(getInt(), x._1.apply())
+      assertEquals(getInt(), loadedGetInt())
+      assertEquals(stringLength(testStr), x._2.apply(testStr))
+      assertEquals(stringLength(testStr), loadedStringLength(testStr))
```

Comment:
Always use known results on the left. If you compare the results of two different things you're testing, they could both be wrong in the same way, and the test would pass:
```suggestion
      assertEquals(42, getInt())
      assertEquals(42, x._1.apply())
      assertEquals(42, loadedGetInt())
      assertEquals(12, stringLength(testStr))
      assertEquals(12, x._2.apply(testStr))
      assertEquals(12, loadedStringLength(testStr))
```